proto/internalapi/scanner/v4/indexer_service.proto

```diff
@@ -36,14 +36,26 @@ message HasIndexReportResponse {
   bool exists = 1;
 }
 
+message GetOrCreateIndexReportRequest {
+  string hash_id = 1;
+  oneof resource_locator {
```

Comment:
I considered moving this `oneof` into an outer message and then using that message in here and `CreateIndexReport`, but that would have caused a bunch of layered structs, and would have been annoying from a user-perspective. I think this is ok, as-is